Add some types

AWare opened the pull request:
This just adds some typescript defs to this. I'm not 100% on how to test this.

AWare committed:
Add some types

AWare commented:
I'm just back off leave, lets get this merged next week.

mbarton commented:
Whoops, I just rewrote the same thing! Will update this PR to include the `types` key in `package.json` (following [this guide](https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html)) and then merge

Commit:
Add types to package.json

This is optional because we are using the right name by convention but recommended

mbarton commented:
Done. Published `0.4.0`
